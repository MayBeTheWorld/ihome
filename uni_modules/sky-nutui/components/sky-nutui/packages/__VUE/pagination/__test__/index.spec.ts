import { mount } from '@vue/test-utils';
import Pagination from '../index.vue';

//测试 totalItems itemsPerPage currentPage @change
test('should render five items and should not emit change event after clicking disabled option', async () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 25,
      itemsPerPage: 5
    }
  });
  expect(wrapper.findAll('.nut-pagination-item')).toHaveLength(5);
});

//测试 mode pageCount
test('should render simple mode', async () => {
  const wrapper = mount(Pagination, {
    props: {
      pageCount: 12,
      mode: 'simple'
    }
  });
  expect(wrapper.findAll('.nut-pagination-item')).toHaveLength(0);
  expect(wrapper.findAll('.nut-pagination-simple')).toHaveLength(1);
});

//测试 forceEllipses showPageSize
test('should render forceEllipses and should emit change event after clicking forceEllipses option', async () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 125,
      showPageSize: 3,
      forceEllipses: true
    }
  });
  expect(wrapper.findAll('.nut-pagination-item')).toHaveLength(4);
});

//测试 click visible
test('should emit change event after clicking visible option', async () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 25,
      itemsPerPage: 5,
      modelValue: 1
    }
  });
  const next = wrapper.find('.nut-pagination-next');
  next.trigger('click');
  expect(wrapper.emitted('update:modelValue')).toBeTruthy();
  expect(wrapper.emitted('change')).toBeTruthy();
});

//测试 click disable
test('should not emit change event after clicking disable option', async () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 25,
      itemsPerPage: 5,
      modelValue: 1
    }
  });
  const prev = wrapper.find('.nut-pagination-prev');
  prev.trigger('click');
  expect(wrapper.emitted('update:modelValue')).toBeFalsy();
  expect(wrapper.emitted('change')).toBeFalsy();
});

//测试slot
test('should render prev-text、next-text slot correctly', () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 50,
      showPageSize: 5
    },
    slots: {
      'prev-text': () => 'Custom PrevText',
      'next-text': () => 'Custom NextText'
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

//测试slot
test('should render page slot correctly', () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 50,
      showPageSize: 5
    },
    slots: {
      page: ({ text = '我呀' }) => `foo${text}`
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});
