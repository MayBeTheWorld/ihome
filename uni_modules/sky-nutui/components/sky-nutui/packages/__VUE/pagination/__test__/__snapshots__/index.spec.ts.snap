// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page slot correctly 1`] = `
"<view class=\\"nut-pagination\\">
  <view class=\\"nut-pagination-prev disabled\\">上一页</view>
  <view class=\\"nut-pagination-contain\\">
    <view class=\\"nut-pagination-item active\\">foo我呀</view>
    <view class=\\"nut-pagination-item\\">foo我呀</view>
    <view class=\\"nut-pagination-item\\">foo我呀</view>
    <view class=\\"nut-pagination-item\\">foo我呀</view>
    <view class=\\"nut-pagination-item\\">foo我呀</view>
  </view>
  <!--v-if-->
  <view class=\\"nut-pagination-next\\">下一页</view>
</view>"
`;

exports[`should render prev-text、next-text slot correctly 1`] = `
"<view class=\\"nut-pagination\\">
  <view class=\\"nut-pagination-prev disabled\\">Custom PrevText</view>
  <view class=\\"nut-pagination-contain\\">
    <view class=\\"nut-pagination-item active\\">1</view>
    <view class=\\"nut-pagination-item\\">2</view>
    <view class=\\"nut-pagination-item\\">3</view>
    <view class=\\"nut-pagination-item\\">4</view>
    <view class=\\"nut-pagination-item\\">5</view>
  </view>
  <!--v-if-->
  <view class=\\"nut-pagination-next\\">Custom NextText</view>
</view>"
`;
